src/main/java/sparksoniq/spark/iterator/flowr/ForClauseSparkIterator.java

```diff
@@ -195,4 +199,66 @@ public void close() {
         }
         return rdd;
     }
+
+
+    @Override
+    public Dataset<Row> getDataFrame(DynamicContext context) {
+        this._df = null;
+
+        if (this._child == null) {
+            // create initial RDD from expression
+            JavaRDD<Item> initialRdd = this.getNewRDDFromExpression(_expression);
```

Comment:
Small bug here. getNewRDDFromExpression should take context as the second parameter, it should not read it from _currentDynamicContext. Maybe it can be fixed in another PR or you can open an issue.